docker/android-system.sh

```diff
@@ -46,64 +79,474 @@ EOF
+    sync bionic
+    sync build
+    sync external/compiler-rt
+    sync external/elfutils
+    sync external/jemalloc
+    sync external/libcxx
+    sync external/libcxxabi
+    sync external/libselinux
+    sync external/mksh
+    sync external/pcre
+    sync external/safe-iop
+    sync external/zlib
+    sync libnativehelper
+    sync prebuilts/clang/linux-x86/host/3.6
+    sync prebuilts/gcc/linux-x86/host/x86_64-linux-glibc2.15-4.8
+    sync prebuilts/misc
+    sync system/core
+
+    case "${ARCH}" in
+        arm)
+            sync prebuilts/gcc/linux-x86/arm/arm-linux-androideabi-4.9
+        ;;
+        arm64)
+            sync prebuilts/gcc/linux-x86/arm/arm-linux-androideabi-4.9
+            sync prebuilts/gcc/linux-x86/aarch64/aarch64-linux-android-4.9
+        ;;
+        x86)
+            sync prebuilts/gcc/linux-x86/x86/x86_64-linux-android-4.9
+        ;;
+        x86_64)
+            sync prebuilts/gcc/linux-x86/x86/x86_64-linux-android-4.9
+        ;;
+    esac
+
+    # avoid build tests
+    rm bionic/linker/tests/Android.mk
+    rm bionic/tests/Android.mk
+    rm bionic/benchmarks/Android.mk
+    # we don't need the relocation packer, and removing
+    # the unittests from it is a bit of work.
+    rm bionic/tools/relocation_packer/Android.mk
+}
+
+# tested on 7.0.0_r36 (SDK 24)
+# tested on 7.1.2_r39 (SDK 25, not supported)
+#   API level 25, requires for Android 7.1, is not provided in NDKs
+android_repo_v7() {
+    sync bionic
+    sync build
+    sync build/kati
+    sync external/boringssl
+    sync external/compiler-rt
+    sync external/elfutils
+    sync external/jemalloc
+    sync external/libcxx
+    sync external/libcxxabi
+    sync external/libselinux
+    sync external/libunwind
+    sync external/libunwind_llvm
+    sync external/llvm
+    sync external/mksh
+    sync external/pcre
+    sync external/safe-iop
+    sync external/zlib
+    sync prebuilts/clang/host/linux-x86
+    sync prebuilts/gcc/linux-x86/host/x86_64-linux-glibc2.15-4.8
+    sync prebuilts/misc
+    sync prebuilts/ndk
+    sync prebuilts/ninja/linux-x86
+    sync system/core
+
+    case "${ARCH}" in
+        arm)
+            sync prebuilts/gcc/linux-x86/arm/arm-linux-androideabi-4.9
+        ;;
+        arm64)
+            sync prebuilts/gcc/linux-x86/arm/arm-linux-androideabi-4.9
+            sync prebuilts/gcc/linux-x86/aarch64/aarch64-linux-android-4.9
+        ;;
+        x86)
+            sync prebuilts/gcc/linux-x86/x86/x86_64-linux-android-4.9
+        ;;
+        x86_64)
+            sync prebuilts/gcc/linux-x86/x86/x86_64-linux-android-4.9
+        ;;
+    esac
+
+    # avoid build tests
+    rm bionic/linker/tests/Android.mk
+    rm bionic/tests/Android.mk
+    rm bionic/benchmarks/Android.mk
+    rm prebuilts/misc/common/android-support-test/Android.mk
+    # we don't need the relocation packer, and removing
+    # the unittests from it is a bit of work.
+    rm bionic/tools/relocation_packer/Android.mk
+
+    remove_tests
```

Comment:
At this point, the Makefiles have test targets inside the same file as the actual products. This means we can't simply remove `Android.mk` files since this might lead to broken inclusions, but worse at the targets and objects declared inside non-test files. This means simple `sed` scripts are no longer viable in all cases.

Reply:
See the comments in the Python module for `android/make.py` for a more detailed description of this.